### ISSUE TITLE
Enable VBA-M Game Boy Link as an Option

### ISSUE THREAD

**Issue by Zombie-Ryushu** [libretro/vbam-libretro#14]
VBA-M's module supports Game Boy Link in WX mode, but only in WX Mode. I think that you should Game Boy Link Mode as a Core Option under RetroArch, and use the underlying VBA-M code to accommodate this

**Comment by retro-wertz** [comment]
thats kinda hard to do atm. even the standalone has its own problems with link and some games. but thats something that's thats gonna happen.

**Comment by Zombie-Ryushu** [comment]
This has been an issue for several years. Do you think going about this the TGB-DUAL Way is more feasible?
Game  Link does work in TGB Dual, But it is only feasible in LAN settings, or Split screen?

**Comment by JackoboLeChocobo** [comment]
I got the GBA linking working successfully using the current code of libretro/vbam-libretro and little modifications :p I have one question for you : I see on GBALink.cpp functions for GB Linking. How does it work ? Someone have tested it with wxvbam ?

**Comment by JackoboLeChocobo** [comment]
So I checked the issues about Link GBA. And I see that --" . https://github.com/visualboyadvance-m/visualboyadvance-m/issues/322#issuecomment-435004198

So much optimism like always in VBA-M team members. So I will continue with my official method of GBA link that works on Windows and Linux computers. Gambatte will be perfect for GB/GBC link. VisualBoyAdvance Link, my project, will be perfect for fully functionnal GBA Linking. If people will be interested by VisualBoyAdvance Link, I will post you the project page.

**Comment by ZachBacon** [comment]
@JackoboLeChocobo as upstream project lead (sadly I wasn't the one that tried to merge your fork in at the time, funny enough back then I was merely a user), if there is anything you could suggest to improve, please let me know. I'd be happy and willing to land any improvements you think are needed.

**Comment by JackoboLeChocobo** [comment]
Question : Where did it go ? https://github.com/visualboyadvance/vbam-libretro . I could access to it yesterday. Can someone confirm the death page on this link ?

**Comment by ZachBacon** [comment]
@JackoboLeChocobo that's the old repo, I've been working with the libretro devs to use the proper upstream https://github.com/visualboyadvance-m/visualboyadvance-m

**Comment by JackoboLeChocobo** [comment]
@ZachBacon Do you prefer to continue the discussion here or on  https://github.com/visualboyadvance-m/visualboyadvance-m with a new ticket ? I have some questions about the code and I will see if I can do something.

**Comment by ZachBacon** [comment]
Go right ahead, I think it would be best since I think this repo either will be archived or removed and upstream recloned for better Pull Request management between downstream and upstream

**Comment by JackoboLeChocobo** [comment]
@ZachBacon, so you want a new ticket on https://github.com/libretro/vbam-libretro/ if I understand ?

**Comment by ZachBacon** [comment]
@JackoboLeChocobo no on https://github.com/visualboyadvance-m/visualboyadvance-m/issues

**Comment by JackoboLeChocobo** [comment]
@ZachBacon Okay thanks ^^

**Comment by ZachBacon** [comment]
@retro-wertz and @rkitover are the other two people, retro-wertz typically looks after the libretro stuff and rkitover and I been mainly taking care of the interface.

**Comment by JackoboLeChocobo** [comment]
@ZachBacon, @retro-wertz, @rkitover and for other people too : go here now to follow this subject  https://github.com/visualboyadvance-m/visualboyadvance-m/issues/356

**Comment by Squall-Leonhart** [comment]
> 1°) VBA-M GBA Link code doesn't run correctly : they tried to use my code of my project VisualBoyAdvance Link project but they had applied it badly ^^"

False, your project was written in JRE, your project was not the basis for BGK's attempts and correcting the link code.

**Comment by JackoboLeChocobo** [comment]
False, my project uses a Jave GUI as command launcher ( I used Java as terminal command to launch the SDL version ) . Nice try.

See here too : https://github.com/visualboyadvance-m/visualboyadvance-m/issues/356#issuecomment-462079583

**Comment by Squall-Leonhart** [comment]
>Nice try : i know my previous versions of MY project better than you.

Obviously not well enough.

The code in vba-m including the basis for bgklink branch was the 1.7.2l source files refactored for 1.8, obviously not working because there were things missing that we never got source code for (1.8l source was never available)

bgk and adamn both worked on their branches individually and it was bgk's changes that got merged back to trunk with many parts working for GBA Linking at the time.

**Comment by JackoboLeChocobo** [comment]
The first versions of VisualBoyAdvance Link used **VisualBoyAdvance 1.7.2 and 1.8.0** ( and noyt VBA-M ) project and denopqrihg's Windows works ( www.vbalink.info/download/V172lsrc.zip ) . Like I said, my project used before a **Java GUI as command launcher** ( I used **Java as terminal command** to launch the **SDL version** that had my GBA Link code ).

Once again, nice try : i know my previous versions of **MY project** better than you.

**Comment by Squall-Leonhart** [comment]
There appears to be a language break down here, nobody is talking about what you used in your project.

We are talking about your continued claims that we used code from yours which you made on the desmume forum in 2017

>Your VBA-M project has a very bad and slow Linux version, and the multiplayer mode doesn't work at all (it wasn't necessary to provide you the code : before, you had tried with the method "copy/paste" , but only a part of the code ... )

**Comment by JackoboLeChocobo** [comment]
Ve can also talk about the fact that you are harassing anyone who does project forks correcting the defects of the project and you can not stand it (both VBA-M and Desmume forks, in my case VisualBoyAdvance Link and Desmume-Reloaded). I'm not alone in finding out ...

https://github.com/visualboyadvance-m/visualboyadvance-m/issues/356#issuecomment-462107416

I don't need to prove to you new things that I already shared and showed here...
If you're a problem with ( French only? ) people for creating better forks, that's not mine.

**Comment by Squall-Leonhart** [comment]
How adorable.  You have a persecution complex.

I don't care what you're doing as long as its not harming anyone or acting against the requirements of a source license.

**Comment by JackoboLeChocobo** [comment]
> I don't care what you're doing as long as **its not harming anyone** or acting against the requirements of a source license.

As your answers ?
![image](https://user-images.githubusercontent.com/42393934/52530428-ff21c380-2d05-11e9-90e4-9e403f246181.png)
![image](https://user-images.githubusercontent.com/42393934/52530431-034de100-2d06-11e9-9752-12180419aabd.png)

**Comment by Squall-Leonhart** [comment]
Considering you have repeatedly pulled me into shit i don't even contribute directly to anymore, yes.

do no harm, respect the gpl.  you call me out for shit the current devs have messed up i'll tell you to fuck off.

**Comment by JackoboLeChocobo** [comment]
![image](https://user-images.githubusercontent.com/42393934/52530870-c507f000-2d0c-11e9-9f19-faeda39c0e1f.png)

Like I said on the ticket of visualboyadvance-m, you are the one who insults me in my case from the start since several years, and you continue always. PDFs of the ngemu forums and from the Github ticket here and visualboyadvance-m will be saved like from the start.

> **do no harm**, respect the gpl. you call me out for shit the current devs have messed up i'll tell you to fuck off.

See https://github.com/libretro/vbam-libretro/issues/14#issuecomment-462107938

**Comment by retro-wertz** [comment]
ugh... why bring this issue here....

**Comment by JackoboLeChocobo** [comment]
You can ask that to Squall here or here too : https://github.com/visualboyadvance-m/visualboyadvance-m/issues/356 . Now it isn't my problem. Like I said, I will continue my works on my fork from today.

**Comment by twinaphex** [comment]
Hey there @JackoboLeChocobo , I just want to say that @ZachBacon feels pretty bad about all this happening and I would like to ask you to give the VBAM team another chance in terms of collaboration. Squall Leonhart is not part of that project anymore and Zach and others would certainly welcome any and all contributions. There can always be disagreements about implementation, but never should things turn vitriolic like this and unprovoked too.

I am sure that if he causes issues again, that either Zach or somebody else with moderator status will take care of it in terms of moderation.

**Comment by JackoboLeChocobo** [comment]
Hi @twinaphex . Thank you for your message. I understand what @ZachBacon feels about the situation. I read what you said about Squall Leonhart and what I understand what you said to me about the collaboration with VBA-M.

I decided from now to restart the collaboration with VBA-M team, to share my works based on denopqrihg works.

@ZachBacon, my code actually doesn't use SFML actually. I will create a new ticket for that and make here a schedule / list of things that i will do, test and apply to ensure a smooth transition of Link support in GBA.

**Comment by simeononsecurity** [comment]
Bump

Would like to see updates on this. I'm surprised with how far the overall project has come that this hasn't been completed just yet.

**Comment by nhobson99** [comment]
Still interested in 2023. I'm so surprised that trading still isn't possible in Retroarch.

**Comment by Zombie-Ryushu** [comment]
It is possible to Trade Pokemon and do other Game Boy and Game Boy Color multiplayer in RetroArch with TGB Dual. You put the Emulator in Split screen mode and then do normal netplay with player 2 as the right game boy, (Super Game Boy excluded, different setup using VBA-M with it seeing a second SNES Controller port) GBA is still not working for this.

**Comment by simeononsecurity** [comment]
> It is possible to Trade Pokemon and do other Game Boy and Game Boy Color multiplayer in RetroArch with TGB Dual. You put the Emulator in Split screen mode and then do normal netplay with player 2 as the right game boy, (Super Game Boy excluded, different setup using VBA-M with it seeing a second SNES Controller port) GBA is still not working for this.

That feature isn't available on certain devices especially ones with small screens. It specifically needs to be added with VBA-M. Anything less is unacceptable.
